aya/src/obj/btf/relocation.rs

```diff
@@ -558,31 +574,41 @@ impl<'a> AccessSpec<'a> {
                 }
             }
             RelocationKind::EnumVariantExists | RelocationKind::EnumVariantValue => match ty {
-                BtfType::Enum(en) => {
+                BtfType::Enum(_) | BtfType::Enum64(_) => {
                     if parts.len() != 1 {
                         return Err(RelocationError::InvalidAccessString {
                             access_str: spec.to_string(),
                         }
                         .into());
                     }
                     let index = parts[0];
-                    if index >= en.variants.len() {
+
+                    let (n_variants, name_offset) = match ty {
```

Comment:
...so then we don't have to match again here.

Reply:
We then get this error:

```rust
error[E0308]: mismatched types
   --> aya-obj/src/btf/relocation.rs:640:53
    |
639 |             RelocationKind::EnumVariantExists | RelocationKind::EnumVariantValue => match ty {
    |                                                                                           -- this expression has type `&types::BtfType`
640 |                 BtfType::Enum(en) | BtfType::Enum64(en) => {
    |                               --                    ^^ expected struct `Enum`, found struct `Enum64`
    |                               |
    |                               first introduced with type `&types::Enum` here
    |
    = note: expected reference `&types::Enum`
               found reference `&types::Enum64`
    = note: in the same arm, a binding must have the same type in all alternatives
```

Not sure if there's a better way to do that in Rust?